### PR TITLE
fix(components): fix Chip, Combobox, and DataList focus states [JOB-82715]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

Upon pulling the DataList and Combobox into Jobber, we found useragent styling on some of our custom-styled `button` and other HTML elements that was being masked by Storybook.

**Chip**

image

**Combobox**
image
image
image

**DataList**
image

It's possible there's a "systemic" fix to this like including a reset for HTML `button` elements in our Atlantis foundations stylesheet, but in service of solving the immediate problem this is my fix.

### Changed

- Changed existing focus background and shadow styling to apply to `focus-visible` contexts for a smarter treatment of focus states: keyboard users always get the full focus treatment, pointer users receive what the browser heuristics determine is appropriate.
- Outline has been removed in contexts where we are already adding a focus style

### Fixed

- useragent styling is replaced with Atlantis styling for focus states

## Testing

- pull prerelease into Jobber or a "sandbox" project

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Chip/Chip.css [hunk 1]
```diff
@@ -50,11 +55,6 @@
   justify-content: center;
 }
 
-.chip .suffix:hover,
```
Comment (chris-at-jobber) [inline packages/components/src/Chip/Chip.css:53]: Chip prefix and suffix do not currently have the ability to receive interactions and should not have a hover state.

**Comment by chris-at-jobber** [comment]
In Jobber, you can verify that the focus styles are as intended, and observe the correct focus behaviour for both keyboard and pointer interactions.

https://github.com/GetJobber/atlantis/assets/39704901/8fdc769e-08fb-478a-97cf-9cfa57ddf454

File: packages/components/src/Chip/Chip.css [hunk 2]
```diff
@@ -24,7 +24,12 @@
   align-items: center;
 }
 
-.chip:focus {
+.chip:focus,
+.chip:focus-visible {
+  outline: none;
```
Comment (chris-at-jobber) [inline packages/components/src/Chip/Chip.css:29]: General note for this PR: I was wondering if just applying `:focus-visible` styling is enough, but in testing at a [basic level](https://codepen.io/murhaus/pen/KKJyJjQ) it _looks like_ there needs to be something telling the element not to receive a focus ring in the `:focus` state even if `:focus-visible` is declared.

Comment (chris-at-jobber) [reply packages/components/src/Chip/Chip.css:29]: image

**Review by scotttjob — APPROVED** [review]
LGTM, does what it says it does on the tin.

**Comment by chris-at-jobber** [comment]
Thanks @scotttjob !

**Pull request merged**